src/impl_methods.rs

```diff
@@ -2444,6 +2444,46 @@ where
         }
     }
 
+    /// Remove the `index`th elements along `axis` and shift down elements from higher indexes.
+    ///
+    /// Decreases the length of `axis` by one.
+    ///
+    /// ***Panics** if `axis` or `index` is out of bounds.
+    pub fn shift_remove_index(&mut self, axis: Axis, index: usize)
+    where
+        S: DataOwned + DataMut,
+    {
+        // TODO: It's possible to choose to shift the elment to the front or the back here,
+        // whichever is closer.
+        let (_, mut tail) = self.view_mut().split_at(axis, index);
+        // shift elements to the back
+        // use swap to keep all elements initialized (as required by owned storage)
+        Zip::from(tail.lanes_mut(axis)).for_each(|mut lane| {
+            let mut lane_iter = lane.iter_mut();
+            let mut dst = if let Some(dst) = lane_iter.next() { dst } else { return };
```

Comment:
If we panic on the `index == self.len_of(axis)` case as I suggested above, the `None` case cannot occur.

Reply:
Yes. Type system wise the existing code is neat enough, don't know if anything else is worthwhile.